hash_practice/exercises.py

```diff
@@ -1,19 +1,103 @@
+import collections
+import heapq
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: While there appears only a single for-loop, 
+        .join() adds O(N) & sorted() adds O(nlogn) during their ideal case

```

Comment:
Note though, that the `n` you refer to is the number of letters in each word, not the number of words in the list (let's call that `m`). Since the O(n * log(n)) `sorted` call gets called `m` times, it would be more accurate to say that it is O(m * n * log(n)).

However, we can make a simplifying assumption. Given that we know the input is a list of English words, and that words in English don't get too large (about 5 letters per word on average), the effect of `n` is going to be dwarfed by the effect of `m`, the number of words in the list. After all, there may be hundreds or thousands of words in the list, but there are not going to be words of that size most likely.